src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+        t_dim: str
+            The name of the time dimension.
+        inplace: bool, optional
+            If True, it will modify the dataarray in place.
+            Otherwise it will return a modified copy.
+
+        Returns
+        -------
+        Union[xarray.DataArray, xarray.DataSet]
+
+        Examples
+        --------
+        >>> import xarray
+        >>> import numpy
+        >>> da = xarray.DataArray(
+        ...  numpy.arange(64).reshape(4,4,4),
+        ...  coords={"lat":numpy.array([87, 88, 89, 90]),
+        ...          "lon":numpy.array([5, 120, 199, 360]),
+        ...          "time":numpy.array([10,11,12,13])}
+        ... )
+        >>> da.aat.set_time_dim(t_dim="time", inplace=True)
+        >>> da.aat.t_dim
+        'time'
+        """
+        data_obj = self._get_obj(inplace=inplace)
+        if t_dim in data_obj.dims:
+            data_obj.aat._t_dim = t_dim
+            return data_obj if not inplace else None
+        raise rioxarray.exceptions.DimensionError(
+            "Time dimension ({t_dim}) not found."
+            f"{_get_data_var_message(data_obj)}"
+        )
+
+    def correct_calendar(
+        self, inplace: bool = False
+    ) -> Union[xarray.DataArray, xarray.DataSet]:
+        """Correct calendar attribute for recognition by xarray.
+
+        Some datasets come with a wrong calendar attribute that isn't
+        recognized by xarray. This function corrects the coordinate
+        attribute to ensure that a ``calendar`` attribute exists
+        and specifies a calendar alias that is supportable by
+        ``xarray.cftime_range``.
+
+        Currently ensures that calendar attributes that are either
+        specified with ``units="months since"`` or ``calendar="360"``
+        explicitly have ``calendar="360_day"``. If and when further
+        issues are found with calendar attributes, support for
+        conversion will be added here.
+
+        Parameters
+        ----------
+        inplace: bool, optional
+            If True, it will modify the dataarray in place.
+            Otherwise it will return a modified copy.
+
+        Returns
+        -------
+        Union[xarray.DataArray, xarray.Dataset]
+            Dataset with transformed calendar coordinate.
+
+        Examples
+        --------
+        >>> import xarray
+        >>> import numpy
+        >>> da = xarray.DataArray(
+        ...  numpy.arange(64).reshape(4,4,4),
+        ...  coords={"lat":numpy.array([87, 88, 89, 90]),
+        ...          "lon":numpy.array([5, 120, 199, 360]),
+        ...          "t":numpy.array([10,11,12,13])}
+        ... )
+        >>> da["t"].attrs["units"] = "months since 1960-01-01"
+        >>> da_crct = da.aat.correct_calendar()
+        >>> da_crct["t"].attrs["calendar"]
+        '360_day'
+        """
+        data_obj = self._get_obj(inplace=inplace)
+        if "calendar" in data_obj[self.t_dim].attrs.keys():
+            if data_obj[self.t_dim].attrs["calendar"] == "360":
```

Comment:
Hmm, so, putting on the same line breaks the formatting rules so then they are auto-formatted with a line break. I think reads better in this manner because of that, although otherwise would put on the same line. What do you think @turnerm? And yeah, it's intentional if second is false, no changes made. Only want to change `360` to `360_day` since `360` isn't a valid calendar recognised by NetCDF (neither are months since). Here's a reference [GitHub issue](https://github.com/Unidata/netcdf4-python/issues/434) that I've now added to the documentation of the function.

Reply:
I would tend towards using an `and`, even if it results in line breaks, because as a rule of thumb you want to reduce nesting (e.g. [flat is better than nested](https://www.python.org/dev/peps/pep-0020/)). 

But my question was really about the `elif` following: Do you mean for it to be `elif` or just an `if`?

Reply:
Ah okay! Will add in. Regarding the `elif`, yeah, would keep that way since the change in the end is the same. I thought about putting both changes on the same line of code, but based on your instructions to use `logger`, I think actually is better to keep separate so the logging information is clearer (one is correcting an incorrect attribute, one is adding an attribute that is equivalent). Let me know what you think.